dvc/scm/__init__.py

```diff
@@ -26,8 +26,34 @@ def SCM(
     Returns:
         dvc.scm.base.Base: SCM instance.
     """
+    from dvc.scm.base import SCMError
+    from dvc.scm.exceptions import SCMError as InternalSCMError
 
-    if no_scm:
-        return NoSCM(root_dir)
+    try:
+        if no_scm:
+            return NoSCM(root_dir)
+        return Git(
+            root_dir, search_parent_directories=search_parent_directories
+        )
+    except InternalSCMError as exc:
+        raise SCMError(str(exc))
 
-    return Git(root_dir, search_parent_directories=search_parent_directories)
+
+def clone(url: str, to_path: str, **kwargs):
+    from .base import CloneError
+    from .exceptions import CloneError as InternalCloneError
+
+    try:
+        return Git.clone(url, to_path, **kwargs)
+    except InternalCloneError as exc:
+        raise CloneError(str(exc))
+
+
+def resolve_rev(scm: "Git", rev: str) -> str:
```

Comment:
Not related to this PR specifically, but we will probably need a `dvc.utils.resolve_rev` function that wraps the separate SCM package's `resolve_rev`.

The existing `scm.git.resolve_rev` method contains DVC specific behavior, where we extend the standard git ref (branch/tag) resolution to also include refs in the exps namespace. This should probably stay in DVC (and not the standalone package).

[`scm.git.__init__.py::resolve_rev`](https://github.com/iterative/dvc/blob/644e9ede924b6416b4ff714fdc0c3463190ab6f2/dvc/scm/git/__init__.py#L305)

Reply:
@pmrowla, this `dvc/scm/__init__.py` will remain as `dvc/scm.py`. So, this `resolve_rev` is the one that will provide DVC specific behaviour. The following contents will be moved here in successive PRs: